src/main/java/com/wolox/training/models/Book.java

```diff
@@ -0,0 +1,142 @@
+package com.wolox.training.models;
+
+import javax.persistence.*;
+
+@Entity
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(nullable = false)
```

Comment:
I recommend using `@javax.validation.constraints.NotNull` instead of `@Column(nullable = false)`, because that way you can check that the value is not null before getting to the database :)

Reply:
By the way, the attribute `genre` is optional :yum: but what Jime says is important with the other attributes. We use the **fail fast** philosophy. If we know something has to fail, it's better to do it as quickly as we can in order to avoid working with non-correct information. So in this case, we should fail as soon as we get the information, without getting to the database.